nimble/host/src/ble_hs_dbg.c

```diff
@@ -149,6 +149,184 @@ ble_hs_dbg_le_event_disp(uint8_t subev, uint8_t len, uint8_t *evdata)
                        get_le16(evdata + 1), evdata[3], evdata[4]);
         break;
 
+    case BLE_HCI_LE_SUBEV_EXT_ADV_RPT:
+    {
+        struct hci_le_subev_ext_adv_rpt *data = (void *) evdata;
+        struct hci_ext_adv_report_param *params = data->params;
+
+        if (len < sizeof(*data) + sizeof(*params)) {
+            BLE_HS_LOG(DEBUG, "Corrupted LE Extended Advertising Report "
+                       "len=%u\n", len);
+            break;
+        }
+
+        BLE_HS_LOG(DEBUG, "LE Extended Advertising Report len=%u num=0x%02x ",
+                   len, data->num_reports);
+
+        for (i = 0; i < data->num_reports; i++) {
+            BLE_HS_LOG(DEBUG, "[%d]={evttype=0x%04x advlen=%u}\n",
+                       i, le16toh(params->evt_type), params->adv_data_len);
+            params += 1;
```

Comment:
you need to adjust this with adv_data_len so params +=1 is not correct here.

Reply:
done